sbin/common/common.sh

```diff
@@ -173,7 +173,30 @@ function setBootJdk() {
     fi
 
     echo "Guessing JDK_BOOT_DIR: ${BUILD_CONFIG[JDK_BOOT_DIR]}"
-    echo "If this is incorrect explicitly configure JDK_BOOT_DIR"
+    echo "=BOOT JDK VERSION OUTPUT="
+    ${BUILD_CONFIG[JDK_BOOT_DIR]}/bin/java -version
+    echo "=/BOOT JDK VERSION OUTPUT="
+
+    # Calculate version number of boot jdk
+    export BOOT_JDK_VERSION_NUMBER=$(${BUILD_CONFIG[JDK_BOOT_DIR]}/bin/java -XshowSettings:properties -version 2>&1 | awk '/java.specification.version/{gsub(/1\./,"",$3);print $3}')
+    if [ -z "$BOOT_JDK_VERSION_NUMBER" ]; then
+      echo "[ERROR] The BOOT_JDK_VERSION_NUMBER was not found. Likelihood is that the boot jdk settings properties don't contain a java.specification.version..."
+      ${BUILD_CONFIG[JDK_BOOT_DIR]}/bin/java -XshowSettings:properties -version 2>&1
+      exit 2
+    fi
+
+    # If bootjdk isn't the same version and isn't a n-1 version, crash out with informational message
+    export REQUIRED_BOOT_JDK=$((${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}-1))
+    # Unless it's jdk8 in which case, a jdk8 boot dir is fine
+    if [ ${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]} -eq 8 ]; then
+      REQUIRED_BOOT_JDK=${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}
+    fi
+
+    if [ ${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]} -ne $BOOT_JDK_VERSION_NUMBER ] && [ $REQUIRED_BOOT_JDK -ne $BOOT_JDK_VERSION_NUMBER ]; then
+      printf "[ERROR] A JDK${BOOT_JDK_VERSION_NUMBER} boot jdk cannot build a JDK${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]} binary\n[ERROR] Please download a JDK${REQUIRED_BOOT_JDK} (preferable) OR JDK${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]} binary and pass it into this script using -J, --jdk-boot-dir"
```

Comment:
This is messy with the printf - can't we just use normal echo statements for this error?

Reply:
I can split it out into multiple echos to make it neater. It's mainly so `/n` is treated as a newline character